how-to/workspace-platform-starter/client/src/modules/lifecycle/example-notification-service/lifecycle.ts

```diff
@@ -0,0 +1,369 @@
+/* eslint-disable @typescript-eslint/brace-style */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import type { WorkspacePlatformModule } from "@openfin/workspace-platform";
+import type {
+	NotificationActionEvent,
+	NotificationClosedEvent,
+	NotificationCreatedEvent,
+	NotificationFormSubmittedEvent,
+	NotificationReminderCreatedEvent,
+	NotificationReminderRemovedEvent,
+	NotificationToastDismissedEvent,
+	NotificationsCountChanged,
+	NotificationOptions
+} from "@openfin/workspace/notifications";
+import type { NotificationClient, NotificationsEventMap } from "workspace-platform-starter/shapes";
+import type {
+	Lifecycle,
+	LifecycleEventMap,
+	LifecycleEvents
+} from "workspace-platform-starter/shapes/lifecycle-shapes";
+import type { Logger, LoggerCreator } from "workspace-platform-starter/shapes/logger-shapes";
+import type { ModuleDefinition, ModuleHelpers } from "workspace-platform-starter/shapes/module-shapes";
+import { isEmpty } from "workspace-platform-starter/utils";
+import type { ExampleNotificationServiceProviderOptions } from "./shapes";
+
+/**
+ * Implementation for the example notification service lifecycle provider.
+ */
+export class ExampleNotificationServiceProvider
+	implements Lifecycle<ExampleNotificationServiceProviderOptions>
+{
+	/**
+	 * The module definition including settings.
+	 * @internal
+	 */
+	private _definition: ModuleDefinition<ExampleNotificationServiceProviderOptions> | undefined;
+
+	/**
+	 * The logger for displaying information from the module.
+	 * @internal
+	 */
+	private _logger?: Logger;
+
+	/**
+	 * Helper methods for the module.
+	 * @internal
+	 */
+	private _helpers: ModuleHelpers | undefined;
+
+	/**
+	 * An object containing current subscriptions.
+	 * @internal
+	 */
+	private _lifeCycleSubscriptions: { [key: string]: LifecycleEvents } | undefined;
```

Comment:
Isn't this back to front, wouldn't you normally key from the LifecycleEvent

Reply:
In this particular setup there can possibly be more than one subscription per type and to avoid the need for arrays (because there shouldn't be many subscriptions and it is an example) we can then loop when unsubscribing all pass the key and value to the unsubscribe notification function.